### DISCUSSION TITLE
Message "Type must be X but is Y" may contain the data field

### DISCUSSION THREAD

**Discussion by tawmoto** [nlohmann/json#2462] [Ideas]
Hi, this is not a bug, but an improvement (i did not find the `improvement` button, sorry)

```
    try
    {
        auto sss = json["garbage1"].get<std::string>();
        auto sss = json["garbage2"].get<std::string>();
        auto sss = json["garbage3"].get<std::string>();
    }
    catch (Json::exception& ex)
    {
        cout << ex.what();
    }
```

Here I get `type must be string but is null`, but it is not clear for which of the fields, maybe it would be useful to have `type must be string for field "garbage2" but is null`
Not sure how hard is this to do, I am trying to debug the project but CLion dies when working with huge files :-D

If this cannot be done, please close the "bug", thanks.

**Comment by gregmarr** [comment]
The problem is that you are calling `.get<std::string>()` on the object returned by `json["garbage"]`.  That object doesn't know anything about the `json` object or that it owns the child object under the name `"garbage"`.  If you had a separate try/catch for each field access, you could tell by where it is caught, but the better thing to do when the input is not trusted to be 100% accurate is to test that `"garbage1"` exists before using it.

**Reply by tawmoto** [reply]
Thanks a lot @gregmarr , it's more clear now
